### PR TITLE
[TS] LPS-150565

### PR DESCRIPTION
**Pull request by dantewang**

### CODE REVIEW

**Comment by catarineaguiar** [comment]
no regressions found

**Comment by vendeltoreki** [comment]
Hi @tinatian ,

With [LPS-150565](https://issues.liferay.com/browse/LPS-150565), we are just introducing some extra controls (configuration values), which the customers can use to limit CPU usage of the portal.

> 1. Who set the expected (max. 3) here? Is the project manager? or it is required by customer?

The customer has a special environment where, as I remember, only 4 threads could run at the same time. Site template propagation background tasks quickly used up these threads, so the server did not answer any requests. However, even if this thread number limit is much higher, it can still be quickly exceeded, because site template propagations can take a lot of time, even hours.
> 2. With your changes, you do not set it to 3, the core thread is set to 5, which is same as before.

The current default values are working for most general usages, these new configurations can help in some special cases, e.g.: when there are a high number of parallel site template propagations.

In summary, we keep the current behavior as the default and introduce some controls over the number of background tasks, and more specifically, site template propagation imports.

**Comment by tinatian** [comment]
@vendeltoreki  Basically, you did not fix the issue in the root, you just add a way to let customer config the best value for them. Please update the ticket description to include this info, I will forward the changes once you updated the ticket.

Thanks.
Tina.

**Comment by vendeltoreki** [comment]
Thanks, @tinatian, I've updated the description of [LPS-150565](https://issues.liferay.com/browse/LPS-150565)

**Comment by tinatian** [comment]
@vendeltoreki Maybe I did not say it clearly, what I mean is to change the "Expected Result" in the ticket description, I do not think it is a valid expected result especially for this "(max. 3) ", you can not have a specific number when you have no idea of the real production system. A (max. 3)  may work with your current customer, but not all.

Also, code conflicts, please rebase.

**Comment by tinatian** [comment]
@vendeltoreki The issue is actually NOT how many thread/task is processed at the same time, it is portal can not respond when a lot of tasks are processing concurrently. The expected result would be : Not block portal when a lot of tasks running

**Comment by vendeltoreki** [comment]
@tinatian , I agree, it doesn't really sound like well defined a bug, but more like a feature request: just making something configurable. Do you think it would be better if we created two new story tickets for this?

E.g.:
* One ticket for the site template propagation import task isolation level (this can be handled by staging team)
* Another ticket for background task executor threadpool settings

**Comment by tinatian** [comment]
@vendeltoreki No, I think it is OK as it is actually a bug, I mean portal did not response when a lot of backgrouond tasks running. Just the expected result is not a specific number of task allowed running concurrent, it is that by configuration you can make portal response while background task running.

Also, please rebase

Tina

**Comment by vendeltoreki** [comment]
Forwarding on: https://github.com/liferay-core-infra/liferay-portal/pull/2660
